src/main/java/team/themoment/hellogsmv3/domain/application/repo/custom/CustomApplicationRepositoryImpl.java

```diff
@@ -0,0 +1,38 @@
+package team.themoment.hellogsmv3.domain.application.repo.custom;
+
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import org.springframework.stereotype.Repository;
+import team.themoment.hellogsmv3.domain.application.entity.abs.AbstractApplication;
+
+import java.util.Optional;
+
+import static team.themoment.hellogsmv3.domain.application.entity.abs.QAbstractApplication.abstractApplication;
+import static team.themoment.hellogsmv3.domain.application.entity.abs.QAbstractMiddleSchoolGrade.abstractMiddleSchoolGrade;
+import static team.themoment.hellogsmv3.domain.application.entity.abs.QAbstractPersonalInformation.abstractPersonalInformation;
+import static team.themoment.hellogsmv3.domain.applicant.entity.QApplicant.applicant;
+
+
+
+
+@Repository
+public class CustomApplicationRepositoryImpl implements CustomApplicationRepository {
+
+    private final JPAQueryFactory queryFactory;
+
+    public CustomApplicationRepositoryImpl(JPAQueryFactory jpaQueryFactory) {
+        this.queryFactory = jpaQueryFactory;
+    }
+
+    @Override
+    public Optional<AbstractApplication> findByApplicantAuthenticationIdWithAllJoins(Long authenticationId) {
+
+        AbstractApplication application = queryFactory.selectFrom(abstractApplication)
+                .join(abstractApplication.middleSchoolGrade, abstractMiddleSchoolGrade).fetchJoin()
+                .join(abstractApplication.personalInformation, abstractPersonalInformation).fetchJoin()
+                .join(abstractApplication.applicant, applicant).fetchJoin()
+                .where(applicant.authenticationId.eq(authenticationId))
+                .fetchOne();
+
+        return Optional.ofNullable(application);
```

Comment:
```suggestion
    public Optional<AbstractApplication> findByApplicantAuthenticationIdWithAllJoins(Long authenticationId) {
        return Optional.ofNullable(queryFactory.selectFrom(abstractApplication)
                .join(abstractApplication.middleSchoolGrade, abstractMiddleSchoolGrade).fetchJoin()
                .join(abstractApplication.personalInformation, abstractPersonalInformation).fetchJoin()
                .join(abstractApplication.applicant, applicant).fetchJoin()
                .where(applicant.authenticationId.eq(authenticationId))
                .fetchOne());
```

인라인화 시켜 바로 반환할 수 있어 보입니다.

Reply:
리뷰 감사합니다.
변수에 값을 할당하고 반환하는 방식이 타입과 변수명을 통해 반환값이 무엇인지 더 쉽게 알수있어 가독성이 더 뛰어나다고 생각하였습니다.

Reply:
저는 메서드명을 보고 반환되는 값이 무엇인지 알 수 있다고 생각하여 리뷰 드렸습니다. 보기에 적절한 방법으로 구현해주시면 감사하겠습니다. 🫡